### PR TITLE
Change clothing shop cart generation to be inside the store

### LINKED ISSUE #51175: Clothing store will only generate shopping cart outside store due to a mistake
### Describe the bug

The shopping cart on the basic `"s_clothes"` map can only be generated on the natural ground outside the store.

This is due to the ASCII character that the map uses for the in-shop floor having been changed from `"."` to `" "` at some point, no doubt for compatibility with the standard building palette, and the shopping cart being checked for generation on the `"."` which used to be shop floor and as per the palette, now is the outside grass.

Most shops define a coordinates zone for cart generation, but a number of maps (probably early in game development ones) all use the `"."` for the main floor and check each one for a low % of generation (typically 1 or 2%). This has advantages and disadvantages.

### Per ASCII character method
Advantage: can strictly control cart location in an irregular area.
Disadvantage: cannot easily control numbers generated or predict likely statistical result of one or more being generated.

### Zone method
Advantage: can strictly control cart numbers placed.
Disadvantage: can only work in a rectangular zone.

### TL;DR:
Anyway, this map uses blanks for floors now, so the line
`"vehicles": { ".": { "vehicle": "shopping_cart", "chance": 1, "status": 0 } }` 
needs to have `"."` changed to `" "`.

### Steps To Reproduce

Place a `"s_clothes"` building and hope you get the base version and not one of the variants (because they don't have carts), then hope that one or more of the 151 `"."` locations rolls the 1% chance and generates one.

### Expected behavior

Like every other store where they generate, I would think that carts should generate inside the store instead of the grass out back.
Not the most convenient spot, as any big box store cart wrangler will tell you.

### Screenshots

TWO of them. Lucky. None IN the store, of course.
image

### Versions and configuration

Win10 tiles x64 Experimental _or_ 0.F-1
English

### Additional context

I'm just going to change it locally and test, then I'll fix it with a PR, probably later today.

### PR DESCRIPTION
**Pull request by RandDenner**
#### Summary
Bugfixes "Fix for mis-generated cart locations in clothing shop map"

#### Purpose of change
Fixes #51175
Changes target terrain with mis-generated cart locations to have the carts generate in the correct places. Makes carts check for in-store generation instead of on the grass outside.

#### Describe the solution
Change the ASCII character potential cart generation is assigned to from a `"."` to `" "`

#### Describe alternatives you've considered
Changing to defined zone method other maps used. Doesn't give significant advantages (discussed in issue #51175).

#### Testing
Changed it locally and spawned a new shop with carts at 100%. Every potential spawn location correctly filled:
image

### CODE REVIEW

**Comment by RandDenner** [comment]
I don't like those carts generating in the northwest and especially the northeast corners.
Replacing those locations with plants is a quick fix.
